### PR TITLE
refactor: feature based separation of bddtests

### LINKED ISSUE #584: bdd tests : revisit packages for steps
currently steps for all tests are placed in same root package under 'test/bdd/'.

Needs to be fixed

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- moved different tests to different packages under `test/bdd/pkg`.
- bddtest framework related code and agent related steps stays under
`test/bdd`.
- cleaned up bdd context and tests to avoid putting test specific data
inside context.
- closes #584

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: test/bdd/agent_sdk_steps.go [hunk 1]
```diff
@@ -7,21 +7,18 @@ SPDX-License-Identifier: Apache-2.0
...
 	"io"
 	"net"
 	"strconv"
-	"strings"
 	"time"
 
 	"github.com/DATA-DOG/godog"
 
-	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/test/bdd/pkg/context"
```
Comment (fqutishat) [inline test/bdd/agent_sdk_steps.go]: extra space

File: test/bdd/bddtests_test.go [hunk 2]
```diff
@@ -16,11 +16,21 @@ import (
 	"time"
 
 	"github.com/DATA-DOG/godog"
+	"github.com/hyperledger/aries-framework-go/test/bdd/pkg/didresolver"
+
```
Comment (fqutishat) [inline test/bdd/bddtests_test.go]: extra space

File: test/bdd/pkg/didresolver/didresolver_steps.go [hunk 3]
```diff
@@ -4,48 +4,50 @@ Copyright SecureKey Technologies Inc. All Rights Reserved.
...
 	"strings"
 	"time"
 
 	"github.com/DATA-DOG/godog"
 
-	"github.com/trustbloc/sidetree-core-go/pkg/document"
-	"github.com/trustbloc/sidetree-core-go/pkg/docutil"
-	"github.com/trustbloc/sidetree-core-go/pkg/restapi/model"
-
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
```
Comment (fqutishat) [inline test/bdd/pkg/didresolver/didresolver_steps.go]: extra space

**Comment by fqutishat** [comment]
After this PR is merged we need to update https://github.com/trustbloc/aries-framework-go-ext

File: test/bdd/pkg/didresolver/didresolver_steps.go (hunk 3 above)
Comment (sudeshrshetty) [reply test/bdd/pkg/didresolver/didresolver_steps.go]: done

File: test/bdd/agent_sdk_steps.go (hunk 1 above)
Comment (sudeshrshetty) [reply test/bdd/agent_sdk_steps.go]: done

File: go.mod [hunk 4]
```diff
@@ -5,6 +5,7 @@
 module github.com/hyperledger/aries-framework-go
 
 require (
+	github.com/DATA-DOG/godog v0.7.13 // indirect
```
Comment (fqutishat) [inline go.mod]: remove it not needed

Comment (sudeshrshetty) [reply go.mod]: done

**Pull request merged**